qiskit_machine_learning/algorithms/trainable_model.py

```diff
@@ -96,9 +97,7 @@ def __init__(
             else:
                 raise QiskitMachineLearningError(f"Unknown loss {loss}!")
 
-        if optimizer is None:
-            optimizer = SLSQP()
-        self._optimizer = optimizer
+        self._optimizer = optimizer or SLSQP()
```

Comment:
Testing against None ought to be explicit . An object is free to implement the dunder bool method so an object could be a valid object i.e. not None yet return False.

Reply:
Yes, it is possible. But optimizers don't redefine bool, there are many places where we do similarly and in another PR I was asked to replace `if` with `or`. I can revert but then we should to check out other places as well.

Reply:
I understand - its just good practice if the test is for None to do it explicitly, as it was before. If I notice a test for None is not done this way I always point it out. Though it works now it shows the intent explicitly and will still work should someone create an optimizer that does implement bool for whatever reason. 

https://peps.python.org/pep-0008/#programming-recommendations
> Also, beware of writing if x when you really mean if x is not None...

Reply:
Rolled back to the explicit `None` check